pkg/clustermarshaller/clustermarshaller.go

```diff
@@ -40,6 +41,13 @@ func MarshalClusterSpec(clusterSpec *cluster.Spec, datacenterConfig providers.Da
 		if err != nil {
 			return nil, fmt.Errorf("failed marshalling resource for cluster spec: %v", err)
 		}
+		if clusterSpec.Spec.ClusterNetwork.DNS.ResolvConf == nil {
```

Comment:
I'm no opposed to this but I'm not sure we need it

Reply:
Without this, it always removes the dns path, even when we specify a resolvConf file.